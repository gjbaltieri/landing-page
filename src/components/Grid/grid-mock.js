// eslint-disable-next-line import/no-anonymous-default-export
export default {
  title: 'Grid',
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Sed hendrerit quam non turpis fringilla, id varius odio aliquet. 
    Curabitur posuere odio et purus consectetur, vel molestie mauris pulvinar. 
    Morbi id lectus id velit suscipit consequat sit amet eget mauris. Nam nec ex ac nunc mattis malesuada. 
    Pellentesque eu libero vitae risus facilisis imperdiet sed sit amet velit.`,
  srcImg: 'https://www.svgrepo.com/show/476982/bridge.svg',
};
