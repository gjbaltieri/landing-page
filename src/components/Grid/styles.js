import styled, { css } from 'styled-components';
import { Title } from '../Header/Title';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.padding.large};
    @media ${theme.media.maxWidth768px} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.padding.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
