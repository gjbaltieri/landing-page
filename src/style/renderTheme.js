import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
