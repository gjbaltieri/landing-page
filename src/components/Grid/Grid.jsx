import P from 'prop-types';
import * as Styled from './styles';
import SectionBackground from '../SectionBackground/SectionBackground';
import Heading from '../Header/Heading';
import TextComponent from '../TextComponent/TextComponent';

export default function Grid({ title, text, srcImg, background = false }) {
  return (
    <SectionBackground background={background}>
      <Styled.Container background={background}>
        <Styled.TextContainer>
          <Heading as='h1' fontSize='xhuge' darkMode={background}>
            {title}
          </Heading>
          <TextComponent>{text}</TextComponent>
        </Styled.TextContainer>
        <Styled.ImageContainer>
          <Styled.Image src={srcImg} alt={title} />
        </Styled.ImageContainer>
      </Styled.Container>
    </SectionBackground>
  );
}

Grid.propTypes = {
  title: P.string.isRequired,
  text: P.string.isRequired,
  srcImg: P.string.isRequired,
  background: P.bool,
};
