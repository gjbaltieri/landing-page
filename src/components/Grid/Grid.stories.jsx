import React from 'react';
import Grid from './Grid';
import Mock from './grid-mock';

export default {
  title: 'Grid',
  component: Grid,
  args: Mock,
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => (
  <div>
    <Grid {...args} />
  </div>
);
