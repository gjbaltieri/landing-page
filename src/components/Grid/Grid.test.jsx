import 'jest-styled-components';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import renderTheme from '../../style/renderTheme';
import Grid from './Grid';
import Mock from './grid-mock';

describe('<Grid/>', () => {
  test('should render Grid with default values', () => {
    const { container } = renderTheme(<Grid {...Mock} />);
    expect(container).toMatchSnapshot();
  });
});
