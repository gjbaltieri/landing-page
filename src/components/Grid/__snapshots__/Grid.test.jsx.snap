// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid/> should render Grid with default values 1`] = `
.c4 {
  color: #333333;
  font-size: 6.2rem;
  font-weight: 900;
  text-transform: none;
}

.c1 {
  padding: 0.8rem;
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c1 .c3 {
  margin-bottom: 4.0rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 100%;
}

.c0 {
  background: #F2F2F2;
  color: #333333;
  -webkit-transition: background 300ms ease-in;
  transition: background 300ms ease-in;
  min-height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3 c4"
          font-size="xhuge"
        >
          Grid
        </h1>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Sed hendrerit quam non turpis fringilla, id varius odio aliquet. 
    Curabitur posuere odio et purus consectetur, vel molestie mauris pulvinar. 
    Morbi id lectus id velit suscipit consequat sit amet eget mauris. Nam nec ex ac nunc mattis malesuada. 
    Pellentesque eu libero vitae risus facilisis imperdiet sed sit amet velit.
        </p>
      </div>
      <div
        class=""
      >
        <img
          alt="Grid"
          class="c6"
          src="https://www.svgrepo.com/show/476982/bridge.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
